modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/impl/DocumentRoutingServiceImpl.java

```diff
@@ -1161,6 +1163,26 @@ public void reassignTask(CoreSession s, final String taskId, final List<String>
         });
     }
 
+    protected Map<String, Serializable> getWorkflowContextualInfo(CoreSession session, Task item) {
+        String workflowInstanceId = item.getProcessId();
+        String nodeId = item.getVariable(DocumentRoutingConstants.TASK_NODE_ID_KEY);
+        Map<String, Serializable> eventInfo = new HashMap<>();
+
+        if (session != null) {
+            GraphNode node = null;
+            if (StringUtils.isNotEmpty(workflowInstanceId)) {
+                NodeAccessRunner nodeAccessRunner = new NodeAccessRunner(session, workflowInstanceId, nodeId);
+                nodeAccessRunner.runUnrestricted();
+                node = nodeAccessRunner.getNode();
+            }
+            if (node != null) {
+                eventInfo = node.getWorkflowContextualInfo(session, true);
```

Comment:
This line could be moved inside previous `if` block, the `node` is never null after `node = nodeAccessRunner.getNode();`. Also, you can directly return the value without assigning it to `eventInfo `.

Once that done, you can remove `GraphNode node = null;` and declare the `node` variable when you get it.
With the early return of `node.getWorkflowContextualInfo` you could return a `Map.of()` (or `Collections.emptyMap()`) for the other return.